### PR TITLE
Uses weight estimate for SIT estimate calculation [delivers #158010863]

### PR DESCRIPTION
**Pull request by stangah**
## Description

We've been using max allowed entitlement for our calculation of estimated SIT reimbursement, mainly because we show that info before a weight estimate is declared. This PR will recalculate SIT reimbursement whenever dependent fields are changed via `PatchPersonallyProcuredMoveHandler`, using the user-declared weight estimate in the calculation.

## Reviewer Notes

~I really hate how this came out, specifically in `updateCalculatedFields`. There's just a lot to do in the patch handler these days, and it's becoming a bit spaghettified. I haven't yet thought of a better way to structure it. Please save me.~ I refactored this a little based on Jim's suggestion, it's a little cleaner.

## Code Review Verification Steps

* [ ] All tests pass.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [ ] Any migrations/schema changes also update the diagram in docs/schema/dp3.sqs
* [ ] There are no aXe warnings for UI.
* [ ] This works in IE.
* Any new client dependencies (Google Analytics, hosted libraries, CDNs, etc) have been:
  * [ ] Communicated to @willowbl00
  * [ ] Added to the list of [network dependencies](https://github.com/transcom/mymove#client-network-dependencies)
* [ ] Request review from a member of a different team.
* [ ] (TRIAL) Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/158010863) for this change

### CODE REVIEW

File: pkg/handlers/personally_procured_move.go [hunk 1]
```diff
@@ -224,33 +224,15 @@ func (h PatchPersonallyProcuredMoveHandler) Handle(params ppmop.PatchPersonallyP
...
-	originPtr := params.PatchPersonallyProcuredMovePayload.PickupPostalCode
-	destinationPtr := params.PatchPersonallyProcuredMovePayload.DestinationPostalCode
-	weightPtr := params.PatchPersonallyProcuredMovePayload.WeightEstimate
-
-	// Figure out if we have values to compare and, if so, whether the new or old value
-	// should be used in the calculation
-	origin, originChanged, originOK := stringForComparison(ppm.PickupPostalCode, originPtr)
-	destination, destinationChanged, destinationOK := stringForComparison(ppm.DestinationPostalCode, destinationPtr)
-	weight, weightChanged, weightOK := int64ForComparison(ppm.WeightEstimate, weightPtr)
+	needsEstimatesRecalculated := h.ppmNeedsEstimatesRecalculated(ppm, params.PatchPersonallyProcuredMovePayload)
```
[the remark this answers is not in the document]
Comment (stangah) [reply pkg/handlers/personally_procured_move.go]: I'm pretty hesitant to try to leverage pointer dereferencing until we remove pointer types from our models. As an example: short of implementing a deep copy method, there's nothing stopping `patchPPMWithPayload` from changing values on `ppm.Advance`, which is a pointer to a `Reimbursement`, and affecting both ppm structs. That wouldn't affect this _particular_ case, but I don't want to start getting in that habit while our code is susceptible to that sort of thing.

File: pkg/handlers/personally_procured_move.go [hunk 2]
```diff
@@ -266,6 +248,41 @@ func (h PatchPersonallyProcuredMoveHandler) Handle(params ppmop.PatchPersonallyP
 
 }
 
+// ppmNeedsEstimatesRecalculated determines whether the fields that comprise
+// the PPM incentive and SIT estimate calculations have changed, necessitating a recalculation
+func (h PatchPersonallyProcuredMoveHandler) ppmNeedsEstimatesRecalculated(ppm *models.PersonallyProcuredMove, patch *internalmessages.PatchPersonallyProcuredMovePayload) bool {
```
[the remark this answers is not in the document]
Comment (tinyels) [reply pkg/handlers/personally_procured_move.go]: We have estimates that are not calculated fields, so perhaps we want to use a different word?

File: pkg/handlers/personally_procured_move.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (tinyels) [reply pkg/handlers/personally_procured_move.go]: It feels like this logic should be in the model not the handler--it's logic related to the model, not the web interface. For instance, if I change the zip on the model in a test, then I would expect the estimates to get updated.

**Comment by tinyels** [comment]
I have some concerns about logic in handlers that I feel belongs in another layer, but I see that has been going on for some time and don't want to block this story for it.

File: pkg/handlers/personally_procured_move.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jim) [reply pkg/handlers/personally_procured_move.go]: @stangah True. I agree with your thinking here.

**Pull request merged**